controllers/barbicanapi_controller.go

```diff
@@ -612,6 +612,36 @@ func (r *BarbicanAPIReconciler) reconcileNormal(ctx context.Context, instance *b
 
 	Log.Info(fmt.Sprintf("[API] Got secrets '%s'", instance.Name))
 
+	// check for  ApplicationCredential
+	acName := fmt.Sprintf("ac-%s", barbican.ServiceName)
+	ac := &keystonev1.ApplicationCredential{}
+	err = r.Client.Get(ctx, types.NamespacedName{Namespace: instance.Namespace, Name: acName}, ac)
+	if k8s_errors.IsNotFound(err) {
+		Log.Info("No ApplicationCredential CR found, using password auth", "ac", acName)
+	} else if err != nil {
+		return ctrl.Result{}, err
+	} else {
+		secretName := ac.Status.SecretName
+		if secretName == "" {
+			Log.Info("ApplicationCredential created but Secret not yet ready, requeueing", "ac", acName)
+			return ctrl.Result{RequeueAfter: 5 * time.Second}, nil
+		}
+		hashAC, res, err := secret.VerifySecret(
+			ctx,
+			types.NamespacedName{Name: secretName, Namespace: instance.Namespace},
+			[]string{"AC_ID", "AC_SECRET"},
+			helper.GetClient(),
+			10*time.Second,
+		)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		if res.RequeueAfter > 0 {
+			return res, nil
+		}
+		configVars["secret-"+secretName] = env.SetValue(hashAC)
```

Comment:
Why do we need this logic?  It appears to be different from all the other secret checks above.